File: frontend/src/types/dr.ts
export interface Dr {
  id: number
  dr_id: string | null

  reported_date: string | null
  closing_date: string | null

  customer_description: string | null
  fault_description: string | null

  detected_by_id: number | null
  repetitions: number | null

  manufacturer_id_number: string | null
  ata: number | null

  close_remarks: string | null
  root_cause_analysis: string | null

  not_our: boolean | null

  priority_id: number | null
  status_id: number | null
  severity_id: number | null

  affected_system_id: number | null

  device_id: number | null
  device_name: string | null

  last_updated: string | null
}

File: backend/app/schemas/dr.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class DrOut(BaseModel):
    model_config = ConfigDict(extra="ignore")

    id: int

    dr_id: str | None = None

    reported_date: datetime | None = None
    closing_date: datetime | None = None

    customer_description: str | None = None
    fault_description: str | None = None

    detected_by_id: int | None = None
    repetitions: int | None = None

    manufacturer_id_number: str | None = None
    ata: int | None = None

    close_remarks: str | None = None
    root_cause_analysis: str | None = None

    not_our: bool | None = None

    priority_id: int | None = None
    status_id: int | None = None
    severity_id: int | None = None

    affected_system_id: int | None = None

    device_id: int | None = None
    device_name: str | None = None

    last_updated: datetime | None = None